srcs/app/server/Connection.cpp

```diff
@@ -41,19 +41,31 @@ SendResponseStatus::Type	Connection::SendResponse() {
 		response_ = response_factory_->Response();
 		raw_response_ = response_->Content();
 		keep_alive_ = response_->KeepAlive();
-		request_->Reset();
+		is_cgi_ = response_->IsCgi();
+		if (is_cgi_) {
+			cgi_output_fd_ = response_->GetCgiOutputFd();
+		}
+		delete response_;
+		response_ = NULL;
```

Comment:
Why do you need to declare additional attributes if you can ask directly to response?

```suggestion
		if (response_->IsCgi();) {
			cgi_output_fd_ = response_->GetCgiOutputFd();
		}
		delete response_;
		response_ = NULL;
```

Reply:
Great idea :100:. I've removed the extra variables.

What do we think about removing the `raw_request_` and `raw_response_` variables and adding the `recv()`/`send()` logic to `IRequest`/`IResponse`?
@Damrod @vdedios

Reply:
In my opinion I would say it make sense to leave `send()` handling in `Connection` class since `IResponse` is an interface whose only responsibility is to build a raw response given a raw request